fiftyone/core/collections.py

```
@@ -923,7 +923,10 @@
             a :class:`fiftyone.core.sample.Sample` or
             :class:`fiftyone.core.sample.SampleView`
         """
-        raise NotImplementedError("Subclass must implement first()")
+        try:
+            return next(iter(self))
+        except StopIteration:
+            raise ValueError("%s is empty" % self.__class__.__name__)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)